EquiCLIP/main_lambda_equitune.py

```diff
@@ -103,13 +103,19 @@ def main(args):
                 print(f"Learning lambda weights: {i}/{args.num_prefinetunes}")
             # zeroshot prediction
             # add weight_net save code for the best model
-
-            # evaluating for only 50 steps using val=True
-            prefinetune_top1_acc, prefinetune_top5_acc, prefinetune_precision, prefinetune_recall, prefinetune_f1_score = eval_clip(
-                args, model, zeroshot_weights, train_loader, val=True, **val_kwargs
-            )
-            wandb.log({"prefinetune_top1_acc": prefinetune_top1_acc, "prefinetune_top5_acc": prefinetune_top5_acc, "prefinetune_precision": prefinetune_precision, 
-                       "prefinetune_recall": prefinetune_recall, "prefinetune_f1_score": prefinetune_f1_score})
+            if args.full_val_pf:
+                prefinetune_top1_acc, prefinetune_top5_acc, prefinetune_precision, prefinetune_recall, prefinetune_f1_score = eval_clip(
+                    args, model, zeroshot_weights, train_loader, val=False, **val_kwargs
+                )
+                wandb.log({"prefinetune_top1_acc": prefinetune_top1_acc, "prefinetune_top5_acc": prefinetune_top5_acc, "prefinetune_precision": prefinetune_precision, 
+                        "prefinetune_recall": prefinetune_recall, "prefinetune_f1_score": prefinetune_f1_score})
+            else:
+                # evaluating for only 50 steps using val=True
+                prefinetune_top1_acc, prefinetune_top5_acc, prefinetune_precision, prefinetune_recall, prefinetune_f1_score = eval_clip(
+                    args, model, zeroshot_weights, train_loader, val=True, **val_kwargs
+                )
+                wandb.log({"prefinetune_top1_acc": prefinetune_top1_acc, "prefinetune_top5_acc": prefinetune_top5_acc, "prefinetune_precision": prefinetune_precision, 
+                        "prefinetune_recall": prefinetune_recall, "prefinetune_f1_score": prefinetune_f1_score})
```

Comment:
avoid code duplication

Reply:
did the change, but might as well skip those arguments altogether and stick with one setting like main.